### PR TITLE
MGMT-18976: Use kube-rbac-proxy for authentication and authorization

### PR DESCRIPTION
**Pull request by alegacy**
This replaces the current authentication and authorization code by leveraging the kube-rbac-proxy.  This reduces the amount of code we must support and allows us to support the standard Kubernetes RBAC mechanism for authorizing users to use the O2IMS API endpoints.

### CODE REVIEW

**Comment by alegacy** [comment]
/hold

...Ok to review, but want to do another pass at testing before I merge

File: internal/controllers/utils/constants.go
Comment (alegacy) [inline internal/controllers/utils/constants.go:150]: @rauhersu  here is where that image is specified.

Comment (rauhersu) [reply internal/controllers/utils/constants.go:150]: Understood, thanks ! We need the same old trick, this needs to be overridden from an env variable defined on a yaml file. I'd need to understand the architecture of your pod better, as I see this kuberbac proxy is executed as an extra container as part of a new pod here: https://github.com/openshift-kni/oran-o2ims/pull/211/files#diff-3cb677c868fecacf955312b938dfa9b44229aee58113a1a07f12fb5a5147631bR777

In any case, if the env we have defined on the CSV is accessible here (I guess you are exec'ing this second pod from the main operator container image) you might just want to grab the environment from the CSV and additionally, should you want to override it as part of the inventory CR - I see this is optional - , define the casuistic here: https://github.com/openshift-kni/oran-o2ims/pull/211/files#diff-3cb677c868fecacf955312b938dfa9b44229aee58113a1a07f12fb5a5147631bR741 which in terms of priorities translates into:

1st image from the CSV (env)
2nd image from the inventory CR
3rd image from the hardcoded golang code.

Note also, that if you want to perform that override choosing your own kuberbac image (2nd), you are entering 'developer mode', bypassing the official customer config.

Cc @donpenney

Comment (rauhersu) [reply internal/controllers/utils/constants.go:150]: /hold

Needs upstream-midstream agreeement

**Comment by alegacy** [comment]
@rauhersu @donpenney I've updated my PR with a new commit to manage the kube-rbac-proxy image differently.  Would you mind taking another look?

File: config/default/kustomization.yaml
Comment (donpenney) [inline config/default/kustomization.yaml:156]: nit: Whitespace is inconsistent with the rest of the yaml

**Review by donpenney — COMMENTED** [review]
/lgtm

File: config/default/kustomization.yaml
Comment (alegacy) [reply config/default/kustomization.yaml]: Fixed.

**Comment by rauhersu** [comment]
Changes lgtm, thx !

/unhold

**Review by donpenney — APPROVED** [review]

**Pull request merged**
